transformer_engine/pytorch/attention/dot_product_attention/backends.py

```diff
@@ -1155,17 +1309,16 @@ def forward(
     @staticmethod
     def backward(ctx, d_out):
         # pylint: disable=missing-function-docstring
-        if ctx.is_output_fp8:
-            assert isinstance(
-                d_out, Float8Tensor
-            ), "Gradient of the DPA output must be in Float8Tensor type for FP8 MHA."
 
-        # FP16/BF16 attn:                  fake_dtype = torch.float16 or torch.bfloat16
-        # FP8 attn, is_output_fp8 = False: fake_dtype = torch.float16 or torch.bfloat16
-        # FP8 attn, is_output_fp8 = True:  fake_dtype = torch.float8_e5m2
-        fake_dtype = d_out.dtype
-
-        d_out = d_out.contiguous()
+        # corner case:
+        # d_out is expected to be in FP8 if is_output_fp8=True,
```

Comment:
That actually is not a corner case - O being in FP8 does not mean that dO will be provided to you in FP8 (and in most cases it will not be).